lib/syscall_shim/uk_prsyscall.c

```diff
@@ -0,0 +1,709 @@
+ */
+static void pr_syscall(struct uk_streambuf *sb, int fmtf,
+		       long syscall_num, long rc, va_list args)
+{
+	switch (syscall_num) {
+#ifdef HAVE_uk_syscall_brk
+	case SYS_brk: {
+			long addr = (long) va_arg(args, long);
+
+			PR_SYSCALL(sb, fmtf, syscall_num,
+					PT_VADDR, addr);
+			PR_SYSRET(sb, fmtf, PT_VADDR, rc);
+		}
+		break;
+#endif /* HAVE_uk_syscall_brk */
+
+#ifdef HAVE_uk_syscall_open
+	case SYS_open:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args, PT_PATH, PT_OFLAGS);
+		PR_SYSRET(sb, fmtf, PT_FD, rc);
+		break;
+#endif /* HAVE_uk_syscall_openat */
+
+#ifdef HAVE_uk_syscall_openat
+	case SYS_openat:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args, PT_DIRFD, PT_PATH,
+			    PT_OFLAGS);
+		PR_SYSRET(sb, fmtf, PT_FD, rc);
+		break;
+#endif /* HAVE_uk_syscall_openat */
+
+#if defined HAVE_uk_syscall_read && defined HAVE_uk_syscall_write
+	case SYS_write:
+		do {
+			int fd    = (int)    va_arg(args, long);
+			void *buf = (void *) va_arg(args, long);
+			__sz len  = (__sz)   va_arg(args, long);
+
+			PR_SYSCALL(sb, fmtf, syscall_num, PT_FD, fd,
+				   PT_BUFP(len), buf, PT_UDEC, len);
+			PR_SYSRET(sb, fmtf, PT_UDEC, rc);
+		} while (0);
+		break;
+	case SYS_read:
+		do {
+			int fd    = (int)    va_arg(args, long);
+			void *buf = (void *) va_arg(args, long);
+			__sz len  = (__sz)   va_arg(args, long);
+
+			PR_SYSCALL(sb, fmtf, syscall_num, PT_FD, fd,
+				   PT_BUFP(((rc >= 0) ? (__sz) rc : len))
+				   | ((rc >= 0) ? PT_OUT : 0x0), buf,
+				   PT_UDEC, len);
+			PR_SYSRET(sb, fmtf, PT_UDEC, rc);
+		} while (0);
+		break;
+#endif /* HAVE_uk_syscall_read && HAVE_uk_syscall_write */
+
+#ifdef HAVE_uk_syscall_stat
+	case SYS_stat:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args,
+			    PT_PATH, PT_STRUCTSTAT);
+		PR_SYSRET(sb, fmtf, PT_STATUS, rc);
+		break;
+#endif /* HAVE_uk_syscall_stat */
+
+#ifdef HAVE_uk_syscall_fstat
+	case SYS_fstat:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args, PT_FD, PT_STRUCTSTAT);
+		PR_SYSRET(sb, fmtf, PT_STATUS, rc);
+		break;
+#endif /* HAVE_uk_syscall_fstat */
+
+#ifdef HAVE_uk_syscall_close
+	case SYS_close:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args, PT_FD);
+		PR_SYSRET(sb, fmtf, PT_STATUS, rc);
+		break;
+#endif /* HAVE_uk_syscall_close */
+
+#ifdef HAVE_uk_syscall_dup
+	case SYS_dup:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args, PT_FD);
+		PR_SYSRET(sb, fmtf, PT_FD, rc);
+		break;
+#endif /* HAVE_uk_syscall_dup */
+
+#ifdef HAVE_uk_syscall_dup2
+	case SYS_dup2:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args, PT_FD, PT_FD);
+		PR_SYSRET(sb, fmtf, PT_FD, rc);
+		break;
+#endif /* HAVE_uk_syscall_dup2 */
+
+#ifdef HAVE_uk_syscall_gettid
+	case SYS_gettid:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args);
+		PR_SYSRET(sb, fmtf, PT_TID, rc);
+		break;
+#endif /* HAVE_uk_syscall_gettid */
+
+#ifdef HAVE_uk_syscall_getpid
+	case SYS_getpid:
+		VPR_SYSCALL(sb, fmtf, syscall_num, args);
+		PR_SYSRET(sb, fmtf, PT_PID, rc);
+		break;
+#endif /* HAVE_uk_syscall_gettid */
```

Comment:
Same here, "getpid" instead of "gettid".

Reply:
:-) :+1: